File: Dot examples/ASP Book Examples/Book Examples/Chapter21/VB/UrlRemapperApp/App_Code/UrlRemapper.vb
Imports System
Imports System.Web
Imports System.Xml
Imports System.Web.Caching
Imports System.Text.RegularExpressions
 
Namespace AspNetUnleashed
 
    Public Class UrlRemapper
        Implements IHttpModule

        Public Sub Init(ByVal app As HttpApplication) Implements IHttpModule.Init
            AddHandler app.BeginRequest, AddressOf app_BeginRequest
        End Sub

        Public Sub app_BeginRequest(ByVal s As Object, ByVal e As EventArgs)
            ' Get HTTP Context
            Dim app As HttpApplication = CType(s, HttpApplication)
            Dim context As HttpContext = app.Context

            ' Get current URL
            Dim currentUrl As String = context.Request.AppRelativeCurrentExecutionFilePath

            ' Get URL Mappings
            Dim urlMappings As XmlDocument = GetUrlMappings(context)

            ' Compare current URL against each URL from mappings file
            Dim nodes As XmlNodeList = urlMappings.SelectNodes("//add")
            For Each node As XmlNode In nodes
                Dim url As String = node.Attributes("url").Value
                Dim mappedUrl As String = node.Attributes("mappedUrl").Value
                If Regex.Match(currentUrl, url, RegexOptions.IgnoreCase).Success Then
                    context.RewritePath(mappedUrl)
                End If
            Next
        End Sub

        Private Function GetUrlMappings(ByVal context As HttpContext) As XmlDocument
            Dim urlMappings As XmlDocument = CType(context.Cache("UrlMappings"), XmlDocument)
            If urlMappings Is Nothing Then
                urlMappings = New XmlDocument()
                Dim path As String = context.Server.MapPath("~/UrlMappings.config")
                urlMappings.Load(path)
                Dim fileDepend As CacheDependency = New CacheDependency(path)
                context.Cache.Insert("UrlMappings", urlMappings, fileDepend)
            End If
            Return urlMappings
        End Function

        Public Sub Dispose() Implements System.Web.IHttpModule.Dispose
        End Sub

    End Class
End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter21/CS/UrlRemapperApp/App_Code/UrlRemapper.cs
using System;
using System.Web;
using System.Xml;
using System.Web.Caching;
using System.Text.RegularExpressions;

namespace AspNetUnleashed
{

    public class UrlRemapper : IHttpModule
    {
        public void Init(HttpApplication app)
        {
            app.BeginRequest += new EventHandler(app_BeginRequest);
        }

        public void app_BeginRequest(Object s, EventArgs e)
        {
            // Get HTTP Context
            HttpApplication app = (HttpApplication)s;
            HttpContext context = app.Context;

            // Get current URL
            string currentUrl = context.Request.AppRelativeCurrentExecutionFilePath;
            
            // Get URL Mappings
            XmlDocument urlMappings = GetUrlMappings(context);

            // Compare current URL against each URL from mappings file
            XmlNodeList nodes = urlMappings.SelectNodes("//add");
            foreach (XmlNode node in nodes)
            {
                string url = node.Attributes["url"].Value;
                string mappedUrl = node.Attributes["mappedUrl"].Value;
                if (Regex.Match(currentUrl, url, RegexOptions.IgnoreCase).Success)
                    context.RewritePath(mappedUrl);
            }
        }

        private XmlDocument GetUrlMappings(HttpContext context)
        {
            XmlDocument urlMappings = (XmlDocument)context.Cache["UrlMappings"];
            if (urlMappings == null)
            {
                urlMappings = new XmlDocument();
                string path = context.Server.MapPath("~/UrlMappings.config");
                urlMappings.Load(path);
                CacheDependency fileDepend = new CacheDependency(path);
                context.Cache.Insert("UrlMappings", urlMappings, fileDepend);

            }
            return urlMappings;
        }

        public void Dispose() { }

    }
}